### PR TITLE
2026 limit questionnaire intent data

### LINKED ISSUE #2026: App force closing when trying to open any forms that have a lot of data passed to them via the Intent
https://github.com/opensrp/fhir-resources/issues/774

**Describe the bug**
Below Forms not loading and are force closing the app:
- Enroll to FP 
- FP visit
- sick child
- pregnancy outcome

**To Reproduce**
- Login to device and sync data.
- Click on a House hold
- Click on a adult female for FP enrollment or Visit
- Click on a child for sick child form

**Expected behavior**
The forms should load to enable us to add data

**Screenshots**
See videos below

https://user-images.githubusercontent.com/84722169/217560613-7002694e-7396-4e9e-847e-09679d447b0b.mp4

**Smartphone (please complete the following information):**

https://user-images.githubusercontent.com/84722169/217560683-336ae9dd-1399-4b2a-85b5-f26bca3b6c80.mp4

Phone: `Samsung A10 android 9`
APK:` quest-v0.1.9-ecbis-rc3-preview.apk`

### PR DESCRIPTION
**Pull request by roywanyaga**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #2026  

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/ActionConfig.kt [hunk 1]
```diff
@@ -39,13 +39,18 @@ data class ActionConfig(
 ) {
   fun paramsBundle(computedValuesMap: Map<String, Any> = emptyMap()): Bundle =
     Bundle().apply {
-      params.map { Pair(it.key, it.value.interpolate(computedValuesMap)) }.forEach {
-        putString(it.first, it.second)
-      }
+      params
+        .filter { it.paramType?.name.equals(PREPOPULATE_PARAM_TYPE) }
+        .map { Pair(it.key, it.value.interpolate(computedValuesMap)) }
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/ActionConfig.kt:44]: I think we can omit this `map` lambda and have the implementation in the `foreach`  below

**Review by ellykits — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/ActionConfig.kt (hunk 1 above)
Comment (roywanyaga) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/ActionConfig.kt:44]: I've updated this to ommit the map

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt [hunk 2]
```diff
@@ -474,6 +458,7 @@ open class QuestionnaireActivity : BaseMultiLanguageActivity(), View.OnClickList
     const val QUESTIONNAIRE_CONFIG = "questionnaire-config"
     const val QUESTIONNAIRE_COMPUTED_VALUES_MAP = "computed-values-map"
     const val QUESTIONNAIRE_ACTION_PARAMETERS = "action-parameters"
+    const val STRING_INTERPOLATION_PREFIX = "@{"
```
Comment (dubdabasoduba) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt:461]: Don't we have this Constant in a universal constants file? I think it's used in several places.

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt (hunk 2 above)
Comment (roywanyaga) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt:461]: I'm not sure whether it is. I cannot seem to trace it. In the S[tring.Interpolate method](https://github.com/opensrp/fhircore/blob/70489e14827789efe8096196e452b8a206a03996/android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/StringExtensions.kt#L42) its specified explicitly

**Review by dubdabasoduba — APPROVED** [review]

**Review by ellykits — APPROVED** [review]

**Pull request merged**